src/app/shared/navbar/searchbar/searchbar.ts

```diff
@@ -0,0 +1,80 @@
+import {Component, ViewChild} from '@angular/core';
+import {MdAutocompleteTrigger, MdSnackBar} from '@angular/material';
+import {Router} from '@angular/router';
+import {FormControl} from '@angular/forms';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/operator/mergeMap';
+
+import {DocumentationItems, DocItem} from '../../documentation-items/documentation-items';
+
+
+@Component({
+  selector: 'search-bar-component',
+  templateUrl: './searchbar.html',
+  styleUrls: ['./searchbar.scss'],
+  host: {
+    '[class.docs-expanded]': '_isExpanded'
+  }
+})
+
+export class SearchBar {
+
+  @ViewChild(MdAutocompleteTrigger)
+  private _autocompleteTrigger: MdAutocompleteTrigger;
+
+  public allDocItems: DocItem[];
+  public filteredSuggestions: Observable<DocItem[]>;
+  public searchControl: FormControl = new FormControl('');
+  public sub;
+
+  public _isExpanded: boolean = false;
+
+  constructor(
+    public _docItems: DocumentationItems,
+    public _router: Router,
+    public _snackBar: MdSnackBar
+  ) {
+    this.allDocItems = _docItems.getAllItems();
+    this.filteredSuggestions = this.searchControl.valueChanges
+      .startWith(null)
+      .map(item => item ? this.filterSearchSuggestions(item) : this.allDocItems.slice());
+  }
+
+  // This handles the user interacting with the autocomplete panel clicks or keyboard.
+  public ngAfterViewInit() {
+    // We listen to the changes on `filteredSuggestions in order to
+    // listen to the latest _autocompleteTrigger.optionSelections
+    this.sub = this.filteredSuggestions
+      .flatMap(_ => this._autocompleteTrigger.optionSelections)
+      .subscribe(evt => this.navigate(evt.source.value.id));
+  }
+
+  public ngOnDestroy() {
+    if (this.sub) { this.sub.unsubscribe(); }
+  }
+
+  public toggleIsExpanded() {
+    this._isExpanded = !this._isExpanded;
+  }
+
+  public displayFn(item: DocItem) {
+    return item.name;
+  }
+
+  public filterSearchSuggestions(searchTerm): DocItem[] {
+    return this.allDocItems.filter(item => new RegExp(`^${searchTerm}`, 'gi').test(item.name));
+  }
+
+  public handlePlainSearch(searchTerm) {
+    const item = this.allDocItems.find(item => item.name.toLowerCase() === searchTerm);
```

Comment:
Partial matches should also just to the first result. E.g., typing "Date" and hitting enter should jump to the datepicker.